File: spec/java/src/io/kaitai/struct/specwrite/TestStrEos.java
package io.kaitai.struct.specwrite;

import io.kaitai.struct.ByteBufferKaitaiStream;
import io.kaitai.struct.ConsistencyError;
import io.kaitai.struct.KaitaiStruct.ReadWrite;
import io.kaitai.struct.testwrite.StrEos;
import org.testng.annotations.Test;

public class TestStrEos extends CommonSpec {
    @Test(expectedExceptions = ConsistencyError.class, expectedExceptionsMessageRegExp = "Check failed: str, expected: 0, actual: 2")
    public void testCheckLongerIo() throws Exception {
        StrEos r = new StrEos();

        r.setStr("Hello");

        r._check();
        r._write(new ByteBufferKaitaiStream(5 + 2));
    }

    @Override
    protected Class<? extends ReadWrite> getStructClass() {
        return StrEos.class;
    }

    @Override
    protected String getSrcFilename() {
        return "term_strz.bin";
    }
}

File: spec/python/specwrite/test_str_eos.py
import io
import unittest
from kaitaistruct import ConsistencyError, KaitaiStream
from specwrite.common_spec import CommonSpec

from testwrite.str_eos import StrEos

class TestStrEos(CommonSpec.Base):
    def __init__(self, *args, **kwargs):
        super(TestStrEos, self).__init__(*args, **kwargs)
        self.struct_class = StrEos
        self.src_filename = 'src/term_strz.bin'

    def test_check_longer_io(self):
        r = StrEos()

        r.str = "Hello"

        r._check()

        with self.assertRaisesRegex(ConsistencyError, u"^Check failed: str, expected: 0, actual: 2$"):
            r._write(KaitaiStream(io.BytesIO(bytearray(5 + 2))))
